### PR TITLE
Add JSON support

### PR DESCRIPTION
**Pull request by drasko**
This commits adds JSON support by adding Jsmn
(http://zserge.com/jsmn.html), a minimalistic JSON parser.

Signed-off-by: Drasko DRASKOVIC <drasko.draskovic@gmail.com>

### CODE REVIEW

File: extras/jsmn/component.mk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Component makefile for extras/onewire
+
+# expected anyone using onewire driver includes it as 'jsmn/jsmn.h'
```
Comment (foogod) [inline extras/jsmn/component.mk]: Couple of comment fixups needed here ("onewire" should be "jsmn")

**Comment by drasko** [comment]
@foogod thanks! Should be fixed now.

**Comment by drasko** [comment]
On the other hand - I did no change in Jsmn code. Maybe this can be introduced as a git submodule?

**Comment by drasko** [comment]
Yep, I just tested the version with submodule, and it works. Would you prefer PR with submodule version?

File: examples/json_test/json_test.c [hunk 2]
```diff
@@ -0,0 +1,91 @@
+/* Very basic example that just demonstrates we can run at all!
```
Comment (projectgus) [inline examples/json_test/json_test.c]: Another minor tweak, this comment is from the original example.

Please add a short header comment & copyright notice for the example being public domain/copyright. Generally the examples are marked public domain / CC0 to make them easy to reuse.

Comment (projectgus) [inline examples/json_test/json_test.c]: (Unless this is from somewhere else, in which case weh ave to keep their copyright/license choice, or unless you'd really like to license it another way.)

File: extras/jsmn/component.mk [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Component makefile for extras/jsmn
+
+# expected anyone using onewire driver includes it as 'jsmn/jsmn.h'
```
Comment (projectgus) [inline extras/jsmn/component.mk]: another 'onewire' typo on this line

File: extras/jsmn/example/jsondump.c [hunk 4]
```diff
@@ -0,0 +1,126 @@
+#include <stdio.h>
```
Comment (projectgus) [inline extras/jsmn/example/jsondump.c]: Please add a header/Copyright notice

File: examples/json_test/json_test.c (hunk 2 above)
Comment (projectgus) [inline examples/json_test/json_test.c]: Scrolling further down I realised this is adapted from the jmsn simple example, so we need to mark its copyright/license accordingly.

**Comment by projectgus** [comment]
Thanks for this @drasko. I added a few minor comments inline.

The git submodule would be preferable, please.

Also, I think there's a solid chance we'll end up with more than one JSON library in the long term. Could you please name the example something like "json_jsmn_simple"?

**Comment by drasko** [comment]
@projectgus thanks for the review. I will close this PR and open new one with fixes and submodule approach.

**Comment by projectgus** [comment]
Great! You don't need to close this PR, you can squash your commits and force push a new version of the PR on top of this one (as described here: http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit/ )

(Or alternatively just keep adding commits, and github's fancy web interface will let me squash them into one commit when I merge the PR.)

**Comment by drasko** [comment]
Fixed and squashed.

**Comment by projectgus** [comment]
This looks good to me except you've somehow picked up two of my commits into the PR when you rebased - not sure how?

@foogod, are you OK with the rest of these changes? I can cherry-pick it into master by hand if need be.

**Comment by foogod** [comment]
Sorry I've been sorta sporadic lately..  Looks fine to me.  Feel free to merge however you think is best..

**Comment by projectgus** [comment]
Thanks again for this, and sorry for the delay. I've cherry-picked it as e7607ffc.

**Pull request closed without merging**